ui/src/errors.ts

```diff
@@ -6,21 +6,37 @@
 import { localize } from "./localize";
 
 // tslint:disable: max-classes-per-file
+export class AzureToolsError extends Error {
+    public suppressDisplay: boolean = false;
+    public suppressReportIssue: boolean = false;
+    constructor(message?: string) {
+        super(message);
+    }
+}
 
-export class UserCancelledError extends Error {
+export class UserCancelledError extends AzureToolsError {
     constructor() {
         super(localize('userCancelledError', 'Operation cancelled.'));
+        this.suppressDisplay = true;
+        this.suppressReportIssue = true;
     }
 }
 
-export class GoBackError extends Error {
+export class GoBackError extends AzureToolsError {
     constructor() {
         super(localize('backError', 'Go back.'));
     }
 }
 
-export class NotImplementedError extends Error {
+export class NotImplementedError extends AzureToolsError {
     constructor(methodName: string, obj: object) {
         super(localize('notImplementedError', '"{0}" is not implemented on "{1}".', methodName, obj.constructor.name));
     }
 }
+
+export class NoResouceFoundError extends AzureToolsError {
+    constructor() {
+        super(localize('noResourcesError', 'No matching resources found.'));
```

Comment:
Seems like we would want to report this error, no?

Reply:
Based on the usage, at least in the docker extension, such error is shown when user tries to pick an item from treeview that doesn't exist, An example is user tries to stop a container, but there were no running container. This doesn't mean bug in the code that require opening the issue. We still show the error, but wouldn't offer the 'Report Issue' button.

Reply:
I would argue the "bug in the code" is that the error is too generic/confusing for the Docker scenario. Ideally Docker could specify a more specific error like "No running containers found."

Reply:
I agree, that the error message is generic and needs to be improved, but not part of this change.

Reply:
@EricJizbaMSFT So do you agree that this should never show the "report an issue" button?

Reply:
I think by default it should be shown and each individual extension can hide it if they want. This is a common error - we've had issues reported on Functions, too, and it's a good reminder that I need to write more specific errors for those scenarios

Reply:
The extension calls ShowTreeItemPicker with certain filter. If no such items found for the given filter, then we show this error. In my opinion, this error should not show 'report issue'. Were there any scenarios in Functions that require user to open issue with this exception?

How would the individual extension overwrite the behavior?